ow_sim_tests/test/arm_check_action.py

```diff
@@ -0,0 +1,294 @@
+         execution that asserts requirements for a successful action operation
+  @param expected_final: Final arm position to be compared with action result
+         (default: None)
+  @param server_timeout: Time the action server is waited for (default: 10.0)
+  @param condition_check_interval: The interval in seconds between calls to
+         action_fail_condition (default: 0.2)
+  @param expected_final_tolerance: allowed distance between result.final and
+         expected_final (default: 0.02)
+  @kwargs: Any properties of goal
+  """
+  def _test_action(self, action_name, action, goal, max_duration,
+                   action_fail_condition,
+                   expected_final = None,
+                   server_timeout = 10.0,
+                   condition_check_interval=0.2,
+                   expected_final_tolerance = 0.02,
+                   **kwargs):
+
+    client = actionlib.SimpleActionClient(action_name, action)
+    connected = client.wait_for_server(timeout=rospy.Duration(server_timeout))
+    self.assertTrue(
+      connected,
+      "Timeout exceeded waiting for %s action server" % action_name
+    )
+
+  # populate goal parameters using kwargs
+    parameter_list = ""
+    for key, value in kwargs.items():
+      if hasattr(goal, key):
+        setattr(goal, key, value)
+        parameter_list += "\n%s : %s" % (key, str(value))
+      else:
+        self.fail(
+          "%s action does not contain parameter %s" % (action_name, key)
+        )
+
+    client.send_goal(goal)
+
+    rospy.loginfo(
+      "\n===%s action goal sent===%s" % (action_name, parameter_list)
+    )
+
+    # monitor for failed conditions during action execution
+    start = rospy.get_time()
+    elapsed = 0.0
+    while not self._is_action_done(client) and elapsed < max_duration:
+      action_fail_condition()
+      rospy.sleep(condition_check_interval)
+      elapsed = rospy.get_time() - start
+
+    self.assertLess(
+      elapsed, max_duration,
+      "Timeout reached waiting for %s action to finish!" % action_name
+    )
+
+    result = client.get_result()
+    rospy.loginfo(
+      "\n===%s action goal sent===%s" % (action_name, parameter_list)
+    )
+    # verify action ended where we expected
+    if expected_final is not None:
+
+      self._assert_point_is_near(
+        result.final, expected_final, expected_final_tolerance,
+        "Arm did not complete the %s action in the position expected! Fianl position is x =%s  y =%s  z =%s" 
+          % (action_name, result.final.x, result.final.y, result.final.z)
+      )
+      
+
+    return result
+
+  def test_01_unstow(self):
+    UNSTOW_MAX_DURATION = 30.0
+    unstow_result = self._test_action(
+      'Unstow',
+      ow_lander.msg.UnstowAction,
+      ow_lander.msg.UnstowGoal(),
+      UNSTOW_MAX_DURATION,
+      self._assert_nothing,
+      expected_final = Point(1.7419, 0.2396, -6.5904),
+      server_timeout = 25.0
```

Comment:
The unstow calls occasionally fail for me when I run the test. I found this to be the case for `terrain_interaction.py` too. I suggest using the same solution as used there, which is to increase the tolerance to 50cm. Be sure to add this to the other two unstow tests too. 
```suggestion
      expected_final_tolerance = 0.5, # unstow requires a really high tolerance
      server_timeout = 25.0
```